modules/runner/src/integrationTest/java/org/apache/ignite/internal/distributionzones/ItDistributionZonesTest.java

```diff
@@ -0,0 +1,258 @@
+import org.apache.ignite.internal.schema.BinaryRowEx;
+import org.apache.ignite.internal.schema.configuration.ExtendedTableConfiguration;
+import org.apache.ignite.internal.schema.configuration.TablesConfiguration;
+import org.apache.ignite.internal.schema.marshaller.TupleMarshallerImpl;
+import org.apache.ignite.internal.table.TableImpl;
+import org.apache.ignite.internal.table.distributed.replicator.TablePartitionId;
+import org.apache.ignite.internal.testframework.WorkDirectory;
+import org.apache.ignite.internal.testframework.WorkDirectoryExtension;
+import org.apache.ignite.internal.util.ByteUtils;
+import org.apache.ignite.table.Tuple;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.Timeout;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+/**
+ * Test suite for rebalance process.
+ */
+@ExtendWith(WorkDirectoryExtension.class)
+@Timeout(90)
+public class ItDistributionZonesTest {
+    private static final IgniteLogger LOG = Loggers.forClass(ItDistributionZonesTest.class);
+
+    /**
+     * Nodes bootstrap configuration pattern.
+     *
+     * <p>rpcInstallSnapshotTimeout is changed to 10 seconds so that sporadic snapshot installation failures still
+     * allow tests pass thanks to retries.
+     */
+    private static final String NODE_BOOTSTRAP_CFG = "{\n"
+            + "  network: {\n"
+            + "    port:{},\n"
+            + "    nodeFinder:{\n"
+            + "      netClusterNodes: [ {} ]\n"
+            + "    }\n"
+            + "  },\n"
+            + "  raft.rpcInstallSnapshotTimeout: 10000"
+            + "}";
+
+    @WorkDirectory
+    private Path workDir;
+
+    private Cluster cluster;
+
+    @BeforeEach
+    void createCluster(TestInfo testInfo) {
+        cluster = new Cluster(testInfo, workDir, NODE_BOOTSTRAP_CFG);
+    }
+
+    @AfterEach
+    @Timeout(60)
+    void shutdownCluster() {
+        cluster.shutdown();
+    }
+
+    @Test
+    @Disabled
+    void assingmentsChangingOnNodeLeaveNodeJoin() throws Exception {
+        cluster.startAndInit(4);
+
+        createTestTable();
+
+        assertTrue(waitAssingments(List.of(
+                Set.of(0, 1, 2),
+                Set.of(0, 1, 2),
+                Set.of(0, 1, 2),
+                Set.of(0, 1, 2)
+        )));
+
+        TableImpl table = (TableImpl) cluster.node(0).tables().table("TEST");
+
+        BinaryRowEx key = new TupleMarshallerImpl(table.schemaView()).marshal(Tuple.create().set("id", 1));
+
+        assertNull(table.internalTable().get(key, new HybridClockImpl().now(), cluster.node(0).node()).get());
+        assertNull(table.internalTable().get(key, new HybridClockImpl().now(), cluster.node(1).node()).get());
+        assertNull(table.internalTable().get(key, new HybridClockImpl().now(), cluster.node(2).node()).get());
+
+        putData();
```

Comment:
It's a bit confusing that on the one hand we expose the key in order to retrieve the data and on the other hand encapsulate it into putData(). Lets be consistent ;)

Reply:
Ok. Fixed it.